benchmarks/newton_solver_benchmark_set/spiegelman_et_al_2016/drucker_prager_compositions.cc

```diff
@@ -0,0 +1,675 @@
+          {
+            sin_phi[c] = std::sin(phi[c] * numbers::PI/180);
+            cos_phi[c] = std::cos(phi[c] * numbers::PI/180);
+          }
+
+        prefactor = get_vector_double("Viscous prefactors",n_fields,prm);
+        stress_exponent = get_vector_double("Stress exponents",n_fields,prm);
+      }
+      prm.leave_subsection();
+
+      prm.enter_subsection("Material model");
+      {
+        prm.enter_subsection ("Drucker prager compositions");
+        {
+
+          // Reference and minimum/maximum values
+          reference_T = prm.get_double("Reference temperature");
+          ref_visc = prm.get_double ("Reference viscosity");
+          min_strain_rate = get_vector_double("Minimum strain rate",n_fields,prm);
+          min_visc = get_vector_double ("Minimum viscosity",n_fields,prm);
+          max_visc = get_vector_double ("Maximum viscosity",n_fields,prm);
+          veff_coefficient = get_vector_double ("Effective viscosity coefficient",n_fields,prm);
+          reference_compressibility  = prm.get_double ("Reference compressibility");
+          use_deviator_of_strain_rate = prm.get_bool ("Use deviator of strain-rate");
+
+
+          // averaging parameters
+          viscosity_averaging_p = prm.get_double("Viscosity averaging p");
+
+          use_analytical_derivative = prm.get_bool("Use analytical derivative");
+
+        }
+        prm.leave_subsection();
+      }
+      prm.leave_subsection();
+
+
+      // Declare dependencies on solution variables
+      this->model_dependence.viscosity = NonlinearDependence::temperature | NonlinearDependence::pressure | NonlinearDependence::strain_rate | NonlinearDependence::compositional_fields;
+      this->model_dependence.density = NonlinearDependence::temperature | NonlinearDependence::pressure | NonlinearDependence::compositional_fields;
+      this->model_dependence.compressibility = NonlinearDependence::none;
+      this->model_dependence.specific_heat = NonlinearDependence::none;
+      this->model_dependence.thermal_conductivity = NonlinearDependence::temperature | NonlinearDependence::pressure | NonlinearDependence::compositional_fields;
+    }
+  }
+}
+
+// explicit instantiations
+namespace aspect
+{
+  namespace MaterialModel
+  {
+    ASPECT_REGISTER_MATERIAL_MODEL(DruckerPragerCompositions,
+                                   "drucker prager compositions",
+                                   " An implementation of a viscous rheology including diffusion"
+                                   " and dislocation creep allowing for different compositions."
+                                   " Designed for testing the Spiegelman et al. (2016) benchmark."
+                                   " Compositional fields can each be assigned individual"
+                                   " activation energies, reference densities, thermal expansivities,"
+                                   " and stress exponents. The effective viscosity is defined as"
```

Comment:
I think much of the documentation below is no longer applicable? Diffusion and dislocation creep is not used in the code above?